src/main/java/org/jenkinsci/plugins/workflow/support/steps/build/BuildUpstreamNodeAction.java

```diff
@@ -9,11 +10,7 @@
 /**
  * Attached to newly-created builds in order to point back to the triggering FlowNode.
  *
- * We annotate the downstream build instead of upstream in order to support no-wait builds,
- * because the downstream run id is not available when they're still queued.
- *
- * Needed for Blue Ocean to annotate the correct step.
- * See <a href="https://issues.jenkins-ci.org/browse/JENKINS-38339">#JENKINS-38339</a>
```

Comment:
Blue Ocean uses `BuildUpstreamCause`, not the action (and fwiw the feature seems to be partially broken based on comments in that tickets and others).

Reply:
AFAICT this is used only by https://github.com/jenkinsci/depgraph-view-plugin/blob/1046773fb05c114b876c09a0fcff7863fbb8da34/src/main/java/hudson/plugins/depgraph_view/model/graph/edge/BuildTriggerEdgeProvider.java#L81-L92 and so could be deprecated in favor of `BuildUpstreamCause` which does essentially the same thing?

Reply:
Yes I think so, I marked it as deprecated in https://github.com/jenkinsci/pipeline-build-step-plugin/pull/127/commits/1a96a2b25711b9f2b6cde3632747747e67faaaf4.